node/lib/util/submodule_util.js

```diff
@@ -358,38 +336,62 @@ exports.getSubmoduleChanges = co.wrap(function *(repo, commit) {
                 const newFile = delta.newFile();
                 const path = newFile.path();
                 if (COMMIT === newFile.mode()) {
-                    result.changed[path] = {
-                        "new": newFile.id().tostrS(),
-                        "old": delta.oldFile().id().tostrS(),
-                    };
-                }
-                else if (SubmoduleConfigUtil.modulesFileName === path) {
-                    result.modulesFileChanged = true;
+                    result[path] = new SubmoduleChange(
+                                                 delta.oldFile().id().tostrS(),
+                                                 newFile.id().tostrS());
                 }
             } break;
             case DELTA.ADDED: {
                 const newFile = delta.newFile();
                 const path = newFile.path();
                 if (COMMIT === newFile.mode()) {
-                    result.added[newFile.path()] = newFile.id().tostrS();
-                }
-                else if (SubmoduleConfigUtil.modulesFileName === path) {
-                    result.modulesFileChanged = true;
+                    result[path] = new SubmoduleChange(null,
+                                                       newFile.id().tostrS());
                 }
             } break;
             case DELTA.DELETED: {
                 const oldFile = delta.oldFile();
                 const path = oldFile.path();
                 if (COMMIT === oldFile.mode()) {
-                    result.removed[oldFile.path()] = oldFile.id().tostrS();
-                }
-                else if (SubmoduleConfigUtil.modulesFileName === path) {
-                    result.modulesFileChanged = true;
+                    result[path] = new SubmoduleChange(oldFile.id().tostrS(),
+                                                       null);
                 }
             } break;
         }
     }
     return result;
+};
+
+/**
+ * Return a summary of the submodule SHAs changed by the specified `commitId` in
+ * the specified `repo`, and flag denoting whether or not the `.gitmodules`
+ * file was changed.
+ *
+ * @asycn
+ * @param {NodeGit.Repository} repo
+ * @param {NodeGit.Commit}     commit
+ * @return {Object}
+ * @return {Object}  return.added    map from path to SHA
+ * @return {Object}  return.changed  map from path to new and old SHAs
+ * @return {Object} return.removed  map from path to SHA
```

Comment:
your formatting here is a bit weird -- do you mean to have two spaces between {Object} and return here?

Reply:
not intended